rust-src/id/src/chain.rs

```diff
@@ -445,6 +444,107 @@ mod tests {
             0,
             policy.clone(),
             &cred_data,
+            &SystemAttributeRandomness {},
+            &Left(EXPIRY),
+        )
+        .expect("Should generate the credential successfully.");
+        let cdi_check = verify_cdi(&global_ctx, &ip_info, &ars_infos, &cdi, &Left(EXPIRY));
+        assert_eq!(cdi_check, Ok(()));
+
+        // Testing with an existing RegId (i.e. an existing account)
+        let existing_reg_id = AccountAddress::new(&cdi.values.cred_id);
+        let cred_data = CredentialData {
+            keys:      {
+                let mut keys = BTreeMap::new();
+                keys.insert(KeyIndex(0), KeyPair::generate(&mut csprng));
+                keys.insert(KeyIndex(1), KeyPair::generate(&mut csprng));
+                keys.insert(KeyIndex(2), KeyPair::generate(&mut csprng));
+                keys
+            },
+            threshold: SignatureThreshold(2),
+        };
+        let (cdi, _) = create_credential(
+            context,
+            &id_object,
+            &id_use_data,
+            1,
+            policy,
+            &cred_data,
+            &SystemAttributeRandomness {},
+            &Right(existing_reg_id),
+        )
+        .expect("Should generate the credential successfully.");
+        let cdi_check = verify_cdi(
+            &global_ctx,
+            &ip_info,
+            &ars_infos,
+            &cdi,
+            &Right(existing_reg_id),
+        );
+        assert_eq!(cdi_check, Ok(()));
+    }
+
+    #[test]
+    fn test_verify_cdi_v1() {
```

Comment:
Can you provide a short summary of what this test does (in the comment above it).